src/coreclr/crossgen-corelib.proj

```diff
@@ -121,6 +121,8 @@
 
     <PropertyGroup>
       <CrossGenDllCmd>$(DotNetCli) @(Crossgen2)</CrossGenDllCmd>
+      <!-- When running under source build, the SDK might alrady target the next major version. -->
+      <CrossGenDllCmd Condition="'$(DotNetBuildFromSource)' == 'true'">$(CrossGenDllCmd) --roll-forward Major</CrossGenDllCmd>
```

Comment:
@ViktorHofer doesn't the `--roll-forward` flag need to be between `$(DotNetCli)` and `@(Crossgen2)`?